### PR TITLE
Documentation migration

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by warden** [comment]
Cool! I've been wanting to do this but you were first ;-) 
Is this somehow integrated into the build pipeline so that there is a documentation preview from branch build available already when the CI build finishes?

**Comment by olamy** [comment]
@warden not really but you can look at the documentation in the branch tree https://github.com/jenkinsci/jira-plugin/tree/documentation_migration

File: README.md [hunk 1]
```diff
@@ -7,6 +7,154 @@ Jenkins JIRA Plugin
 
 Check documentation at [JIRA plugin page](https://plugins.jenkins.io/jira).
 
+
+Older versions of this plugin may not be safe to use. Please review the
+following warnings before using an older version:
+
+-   [CSRF vulnerability and missing permission checks allowed capturing credentials](https://jenkins.io/security/advisory/2018-09-25/#SECURITY-1029){.external-link}
+-   [Folder-scoped Jira sites are able to access System-scoped credentials](https://jenkins.io/security/advisory/2019-11-21/#SECURITY-1106){.external-link}
```
Comment (dwnusbaum) [inline README.md]: You should delete these warnings, they are automatically shown on the sidebar of the plugin site (e.g. [script-security](https://plugins.jenkins.io/script-security)).

File: README.md [hunk 2]
```diff
@@ -7,6 +7,154 @@ Jenkins JIRA Plugin
...
+  
+**TOC:**
+
+#### Using JIRA REST API
+
+This plugin has an optional feature to update JIRA issues with a back
+pointer to Jenkins build pages. This allows the submitter and watchers
+to quickly find out which build they need to pick up to get the fix.
+
+![](docs/images/Plugin_Configuration.jpg){.confluence-embedded-image}
```
Comment (dwnusbaum) [inline README.md]:
```suggestion
![](docs/images/Plugin_Configuration.jpg)
```

File: README.md [hunk 3]
```diff
@@ -7,6 +7,154 @@ Jenkins JIRA Plugin
...
+By taking advantages of Jenkins'
+[fingerprint](https://wiki.jenkins.io/display/JENKINS/Fingerprint)
+feature, when your other projects that depend on this project pick up a
+build with a fix, those build numbers can also be recorded in JIRA.
+
+This is quite handy when a bug is fixed in one of the libraries, yet the
+submitter wants a fix in a different project. This happens often in my
+work, where a bug is reported against JAX-WS but the fix is in JAXB. 
+
+For curious mind, see [this thread for how this works behind the scene](http://jenkins.361315.n4.nabble.com/How-can-does-Hudson-Jira-integration-works-td374680.html){.external-link}.
```
Comment (dwnusbaum) [inline README.md]:
```suggestion
For curious mind, see [this thread for how this works behind the scene](http://jenkins.361315.n4.nabble.com/How-can-does-Hudson-Jira-integration-works-td374680.html).
```

File: README.md (hunk 1 above)
Comment (olamy) [reply README.md]: good catch done

File: README.md [hunk 4]
```diff
@@ -5,7 +5,142 @@ Jenkins JIRA Plugin
...
+
+If you also want to use this feature, you need to supply a valid user
+id/password. If you need the comment only to be visible to a certain
+JIRA group, e.g. *Software Development*, enter the groupname. 
+
+Now you also need to configure jobs. I figured you might not always have
+write access to the JIRA (say you have a Jenkins build for one of the
+Apache commons project that you depend on), so that's why this is
+optional.  
+![](http://weblogs.java.net/blog/kohsuke/archive/20070312/jobConfig.png)
```
Comment (halkeye) [inline README.md]: FYI This is a dead link

Comment (olamy) [reply README.md]: @halkeye thanks buddy good catch!

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
+Changelog
```
Comment (artkoshelev) [inline CHANGELOG.md]: is it auto-generated?

**Pull request merged**